app/views/clans/ClansView.coffee

```diff
@@ -52,12 +43,17 @@ module.exports = class ClansView extends RootView
       @refreshNames @publicClans.models
       @render?()
     @supermodel.loadCollection(@publicClans, 'public_clans', {cache: false})
+    @publicClans = _.filter(@publicClans.models, (clan) -> clan.get('type') is 'public')
+
     @myClans = new CocoCollection([], { url: "/db/user/#{me.id}/clans", model: Clan, comparator: sortClanList })
     @listenTo @myClans, 'sync', =>
       @refreshNames @myClans.models
       @render?()
     @supermodel.loadCollection(@myClans, 'my_clans', {cache: false})
+    @myClans = @myClans.models
+
```

Comment:
This logic that was in `getRenderData` should go into a `onLoaded` function. Right now the logic runs after the `myClans` and `publicClans` collections have been created, but before their network requests have resolved. So they're empty at this point. Instead, put the logic in `onLoaded` like this:

```coffeescript
onLoaded: ->
  # prepare loaded data here
  super()
```

Any data manipulation happens after the data's been loaded, but before the page is re-rendered.

Also, view properties should not change in type, and setting `@myClans = @myClans.models` and `@publicClans = _.filter(@publicClans.models, ...)` changes those properties from collections to arrays. Either the view should use the collections (in which case `@publicClans` should be `reset` with the filtered models array, and the view should be updated to use collections) or the arrays should be set to new properties (`@publicClansArray` and `@myClansArray` for example).

Really appreciate you going through and standardizing views!